zebra-consensus/src/checkpoint.rs

```diff
@@ -155,4 +152,231 @@ where
     }
 }
 
-// TODO(teor): tests
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use color_eyre::Report;
+    use eyre::{bail, ensure, eyre};
+    use tower::{util::ServiceExt, Service};
+
+    use zebra_chain::serialization::ZcashDeserialize;
+
+    #[tokio::test]
+    #[spandoc::spandoc]
+    async fn checkpoint_single_item_list() -> Result<(), Report> {
+        let block0 =
+            Arc::<Block>::zcash_deserialize(&zebra_test_vectors::BLOCK_MAINNET_GENESIS_BYTES[..])?;
+        let hash0: BlockHeaderHash = block0.as_ref().into();
+
+        // Make a checkpoint list containing only the genesis block
+        let genesis_checkpoint_list: HashMap<BlockHeight, BlockHeaderHash> =
+            [(block0.coinbase_height().unwrap(), hash0)]
```

Comment:
It might be worth pulling in a crate to provide a `map!` macro for simple cases like this

Reply:
We'd only need it during testing. Or maybe for the hard-coded genesis block hashes.

Reply:
Actually, it looks like we need to initialise the state with the relevant genesis block. So we'd only need to mandatory sapling hashes.

See https://github.com/ZcashFoundation/zebra/issues/504#issuecomment-647667834